sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -594,10 +591,15 @@ public static TableRow toTableRow(Row row) {
           java.time.format.DateTimeFormatter localDateTimeFormatter =
               (0 == localDateTime.getNano()) ? ISO_LOCAL_DATE_TIME : BIGQUERY_DATETIME_FORMATTER;
           return localDateTimeFormatter.format(localDateTime);
-        } else if ("Enum".equals(identifier)) {
+        } else if (EnumerationType.IDENTIFIER.equals(identifier)) {
           return fieldType
               .getLogicalType(EnumerationType.class)
               .toString((EnumerationType.Value) fieldValue);
+        } else if (NanosInstant.IDENTIFIER.equals(identifier)) {
+          if (fieldValue instanceof java.time.Instant) {
```

Comment:
@amuletxheart I think we should align with other field types and just perform the type coercion without the if clause here. This allows us to fail loudly instead of silently dropping the field.